### PR TITLE
Add `--ncores` param to clean clients

### PR DESCRIPTION
**Pull request by joaomcteixeira**
You are about to submit a new Pull Request. Before continuing make sure you read the [contributing guidelines](CONTRIBUTING.md) and you comply with the following criteria:

- [x] You have stick to Python. Talk with us before adding other programming languages to HADDOCK3
- [ ] Your PR is about CNS
- [x] Your code is well documented: proper docstrings and explanatory comments for those tricky parts
- [x] You structured the code into small functions as much as possible. You can use classes if there's a (state) purpose
- [x] code follows our coding style
- [x] You wrote tests for the new code
- [x] `tox` tests pass. *Run `tox` command inside the repository folder*
- [x] `-test.cfg` examples execute without errors. *Inside `examples/` run `python run_tests.py -b`*
- [x] PR does not add any *install dependencies* unless permission granted by the HADDOCK team
- [x] PR does not break licensing
- [ ] Your PR is about writing documentation for already existing code :fire:
- [ ] Your PR is about writing tests for already existing code :godmode:

---

Add the `--ncores` parameter to the `clean` and `unpack` clients.

### CODE REVIEW

File: src/haddock/clis/cli_clean.py [hunk 1]
```diff
@@ -9,19 +9,22 @@
...
+specific folder of the workflow step. <ncores> defined the number of
+threads to use.
 
 Usage::
 
     haddock3-clean -h
     haddock3-clean <run_directory>
     haddock3-clean run1
     haddock3-clean run1/1_rigidbody
+    haddock3-clean run1 -n  # uses all cores
```
Comment (amjjbonvin) [inline src/haddock/clis/cli_clean.py:21]: I assume that without the -n argument all cores are used as well, correct?

File: src/haddock/libs/libcli.py [hunk 2]
```diff
@@ -40,6 +40,24 @@ def add_rundir_arg(ap):
...
 
 
+def add_ncores_arg(ap):
+    """Add number of cores option."""
+    ap.add_argument(
+        "-n",
+        "--ncores",
+        dest='ncores',
+        help=(
+            "The number of threads to use. Uses 1 if not specified. "
```
Comment (amjjbonvin) [inline src/haddock/libs/libcli.py:50]: Ok - that answers my previous questions :-)  All good!

**Review by amjjbonvin — APPROVED** [review]

File: src/haddock/libs/libcli.py (hunk 2 above)
Comment (joaomcteixeira) [reply src/haddock/libs/libcli.py:50]: Yes. `haddock3-clean -h` will tell also :wink:

File: src/haddock/clis/cli_clean.py (hunk 1 above)
Comment (joaomcteixeira) [reply src/haddock/clis/cli_clean.py:21]: the `-n` alone will use all cores. withOUT `-n` only 1 will be used.

**Comment by joaomcteixeira** [comment]
Once the workflow tests are done, I will merge :+1:

**Comment by amjjbonvin** [comment]
Swift action!

**Pull request merged**